cloud/softlayer/softlayer.py

```diff
@@ -0,0 +1,428 @@
+      datacenter: dal09
+      tag: ansible-module-test
+      hourly: True
+      private: False
+      dedicated: False
+      local_disk: True
+      cpus: 1
+      memory: 1024
+      disks: [25]
+      os_code: UBUNTU_LATEST
+
+- name: Build additional instances
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Build instances request
+    local_action:
+      module: softlayer
+      quantity: 2
+      increment: True
+      increment_start: 2
+      hostname: instance
+      domain: anydomain.com
+      datacenter: dal09
+      tag: ansible-module-test
+      hourly: True
+      private: False
+      dedicated: False
+      local_disk: True
+      cpus: 1
+      memory: 1024
+      disks: [25,100]
+      os_code: UBUNTU_LATEST
+      image_id:
+      private_vlan:
+      ssh_keys: []
+      post_uri:
+
+- name: Wait for instances
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Wait by tag
+    local_action:
+      module: softlayer
+      state: wait
+      tag: ansible-module-test
+
+- name: Cancel instances
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Cancel by tag
+    local_action:
+      module: softlayer
+      state: absent
+      tag: ansible-module-test
+'''
+
+import time
+from ansible.module_utils.facts import *
+
+
+DATACENTERS = ['ams01','ams03','dal01','dal05','dal06','dal09','fra02','hkg02','hou02','lon2','mel01','mex01','mil01','mon01','par01','sjc01','sjc03','sao01','sea01','sng01','syd01','tok02','tor01','wdc01','wdc04']
+CPU_SIZES = [1,2,4,8,16]
+MEMORY_SIZES = [1024,2048,4096,6144,8192,12288,16384,32768,49152,65536]
+INITIALDISK_SIZES = [25,100]
+LOCALDISK_SIZES = [25,100,150,200,300]
+SANDISK_SIZES = [10,20,25,30,40,50,75,100,125,150,175,200,250,300,350,400,500,750,1000,1500,2000]
+NIC_SPEEDS = [10,100,1000]
+
+try:
+
+  import SoftLayer
+  from SoftLayer import VSManager, ISCSIManager, ImageManager, Client
+
+  HAS_SL = True
+
+  client = Client()
+  vsManager = VSManager(client)
+
+except ImportError:
+  HAS_SL = False
+
+
+
+def create_virtual_instances(module):
+
+  increment = module.params.get('increment')
+  quantity = module.params.get('quantity')
+  increment_start = module.params.get('increment_start')
+
+  instances = []
+  counter = 1
+  total_instances = quantity
+
+  if increment:
+    hostname = module.params.get('hostname')
+    if increment_start > 1:
+      counter = increment_start
+      total_instances = total_instances + (counter-1)
+
+    while counter <= total_instances:
+      module.params['hostname'] = hostname + "-" + str(counter)
+      (changed, instance) = create_virtual_instance(module)
+      instances.append(instance)
+      counter += 1
+  else:
+    while counter <= total_instances:
+      (changed, instance) = create_virtual_instance(module)
+      instances.append(instance)
+      counter += 1
+
+  return True, instances
+
+
+def create_virtual_instance(module):
+
+  hostname = module.params.get('hostname')
```

Comment:
most of these are just then reassigned down below, wouldn't it just be simpler to directly use the module.params?